### ISSUE TITLE
Uncaught TypeError: Cannot read property 'set' of undefined

### ISSUE THREAD

**Issue by loranger** [randy3k/remote-atom#24]
[Enter steps to reproduce below:]

1. ssh server
2. rmate a/simple/file
3. Try again a few times and the error disappear

**Atom Version**: 1.3.1
**System**: Mac OS X 10.11.2
**Thrown From**: [remote-atom](https://github.com/randy3k/remote-atom) package, v1.3.6

### Stack Trace

Uncaught TypeError: Cannot read property 'set' of undefined

```
At /Users/loranger/.atom/packages/remote-atom/lib/remote-atom.coffee:133

TypeError: Cannot read property 'set' of undefined
    at Session.parse_setting (/Users/loranger/.atom/packages/remote-atom/lib/remote-atom.coffee:133:18)
    at Session.parse_line (/Users/loranger/.atom/packages/remote-atom/lib/remote-atom.coffee:128:14)
    at Session.parse_chunk (/Users/loranger/.atom/packages/remote-atom/lib/remote-atom.coffee:115:18)
    at Socket.<anonymous> (/Users/loranger/.atom/packages/remote-atom/lib/remote-atom.coffee:99:14)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)
```

### Commands

```
     -5:45.2.0 editor:select-to-first-character-of-line (atom-text-editor.editor.is-focused)
  2x -5:44.9.0 core:backspace (atom-text-editor.editor.is-focused)
     -5:43.3.0 core:save (atom-text-editor.editor.is-focused)
  2x -5:42.4.0 core:move-up (atom-text-editor.editor.is-focused)
 10x -5:41.5.0 core:move-down (atom-text-editor.editor.is-focused)
     -5:39.8.0 core:move-up (atom-text-editor.editor.is-focused)
  2x -5:39.5.0 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
  4x -5:38.9.0 core:move-right (atom-text-editor.editor.is-focused)
  4x -5:38.1.0 core:select-right (atom-text-editor.editor.is-focused)
     -5:37.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -5:35.7.0 core:save (atom-text-editor.editor.is-focused)
     -3:53.6.0 editor:consolidate-selections (atom-text-editor.editor.mini.is-focused)
     -3:53.6.0 core:cancel (atom-text-editor.editor.mini.is-focused)
  4x -3:51.2.0 pane:show-next-item (atom-text-editor.editor.is-focused)
     -2:19 linter:next-error (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-gruvbox.theme-nucleus-dark-ui)
     -2:15 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "nucleus-dark-ui",
      "gruvbox"
    ],
    "ignoredNames": [
      ".git",
      ".svn",
      ".DS_Store",
      ".sublime-project",
      ".sublime-workspace",
      "._"
    ],
    "packagesWithKeymapsDisabled": [
      "terminal-plus"
    ],
    "disabledPackages": [
      "terminal-plus",
      "symbols-view"
    ]
  },
  "remote-atom": {
    "launch_at_startup": true,
    "keep_alive": true
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.3.0
aligner, v0.17.4
atom-beautify, v0.28.19
atom-ctags, v4.4.2
auto-detect-indentation, v0.4.2
autoclose-html, v0.19.0
build, v0.50.0
build-gulp, v0.5.0
editor-stats, v0.17.0
emmet, v2.3.17
file-icons, v1.6.13
gruvbox, v1.0.5
highlight-selected, v0.11.1
language-blade, v0.16.0
language-crontab, v0.1.0
language-nginx, v0.6.1
laravel, v0.7.0
less-than-slash, v0.11.1
linter, v1.11.3
linter-csslint, v1.1.0
linter-htmlhint, v0.2.1
linter-jshint, v1.2.1
linter-jsonlint, v1.1.2
linter-php, v1.1.6
linter-sass-lint, v0.4.3
minimap, v4.18.0
minimap-find-and-replace, v4.4.0
minimap-highlight-selected, v4.3.1
minimap-pigments, v0.1.7
nucleus-dark-ui, v0.6.14
php-twig, v4.0.0
pigments, v0.19.3
remote-atom, v1.3.6

# Dev
No dev packages
```

**Comment by randy3k** [comment]
The error occurred because `@file` in [line 133](https://github.com/randy3k/remote-atom/blob/fb8de8ad8f0b47f2d1c57fc8fe4772b24269faef/lib/remote-atom.coffee#L133) in empty. However, I have no idea when it happened, [line 125](https://github.com/randy3k/remote-atom/blob/fb8de8ad8f0b47f2d1c57fc8fe4772b24269faef/lib/remote-atom.coffee#L125) should have initialized the `@file` when the "open" command is received.

**Comment by randy3k** [comment]
Which rmate client were you using?

**Comment by loranger** [comment]
Is there a way to get the @file value as a debug ?
I use the official rmate client installed with ruby gems (rmate 1.5.8)

**Comment by hlvnst** [comment]
**tl;dr**: Workaround: Use the [Bash version of `rmate`][rmate-bash].

I'm having the same issue, and I'm pretty sure that the cause is either in the [official `rmate` Ruby implementation][rmate-ruby], or in the interaction between it and this package.

Being TextMate user, I decided to take Atom for a spin, and one of the first things I did was to see if there's anything like `rmate` for Atom, which led me to this package. Therefore, this is the only package I have installed, thus ruling out any possible conflicts.

Because I have the exact same version of `rmate` as @loranger on my server, I decided to try the [Bash version][rmate-bash] instead. After many attempts of opening files (with some pauses in between), I can safely say this issue doesn't exist when using the Bash implementation.

Seeing that I'm pretty new to Atom and my Ruby skills are nonexistent, fixing this issue is beyond me, but at least this narrows it down a bit.

[rmate-bash]: https://github.com/aurora/rmate
[rmate-ruby]: https://github.com/textmate/rmate

**Comment by randy3k** [comment]
I still cannot reproduce this with the official ruby rmate in the [master](https://github.com/textmate/rmate) branch. 

@loranger and @hlvnst , could you try to update to the latest version of it and try?

I myself also use the bash version, so it would be easier for me to pick up any issues on the bash version  rmate versus other versions of rmate.

**Comment by hlvnst** [comment]
Apart from some changes to support files, the master branch hasn't diverged from the latest release. So as I expected, no change.

**Comment by randy3k** [comment]
I might figured it out. It may be due to how the command was sent from the remote server, specifically,
the open command may be received as "open" or "open\n" with the trailing new line char.

In the origin code, it is assume that it always comes with a new line char. However, occasionally, the connection is randomly cut into chunks which "open\n" may become "open" then "\n". That's why the error is so random.

2d879ae should address the issue. Remote-atom 1.3.9 should be ready to be installed on your machine.

**Comment by loranger** [comment]
I just installed the update. I'll use it today and let you know.
Thank you @hlvnst for the workaround.
I am a textmate switcher too, and remote-atom was the package that convinced me to test atom so, thank you so much, @randy3k !

**Comment by hlvnst** [comment]
Version 1.3.9 seems to have indeed fixed the issue. I can no longer reproduce it on my setup.

**Comment by loranger** [comment]
I didn't had a lot of files to edit remotely, but I didn't notice any issue either with 1.3.9
Let's close this ticket !
